cybered/hashing/templates/hashing/motivation.html

```diff
@@ -0,0 +1,94 @@
+            piece of information. For example, Alice might use a <b>hash function</b> to generate the following
+            digest to describe her long message:
+
+            <pre>5578833a0c6cb26394a1414140718cab</pre>
+
+            Instead of having to tell Bob the whole message in-person, she can just tell him that short piece of
+            information. Then, when Bob gets Alice's message, he can use the same hash function to generate a message
+            digest. If the message digest that Bob gets is the same as the one Alice gave him, then it is <i>very</i>
+            likely that he received the correct message!
+        </p>
+
+        <h3>Hash Functions</h3>
+        <p>
+            Hash functions take a message and produce a message digest from it. There are many ways that a
+            hash function can do this, but the most important feature of a hash function is that it is a one-way
+            function, which means you can always get the same digest from a
+            specific input, but you can never start with the digest and figure out the original input.
+        </p>
+        <p>
+            A very simple hash function might convert every letter to a number, and then add them all up.
+            Lets say that A=1, B=2, C=3...
+            <ul>
+                <li>cat = 3 + 1 + 20 = 24</li>
+                <li>dog = 4 + 15 + 7 = 26</li>
+            </ul>
+        </p>
+        <p>
+            Because hash functions are used to condense data, it is possible for multiple messages
+            to result in the same digest. This is called a <b>collision</b> and there's nothing wrong
+            with it, as long as that is very difficult to do on purpose. If Chuck is able to make
+            a new message with the same digest as Alice's, then he can send that to Bob, and Bob
+            will think it is the real message!
+        </p>
+        <p>
+            Is our hash function above a good hash function according to this rule?
+            No, it's not! We can come up with many words that have the same digest. The easiest
+            way is to re-arrange the letters of the message.
+            <ul>
+                <li>dog = 4 + 15 + 7 = 26</li>
+                <li>god = 7 + 15 + 4 = 26</li>
+            </ul>
+        </p>
+
+        <p> Modern hash functions can be very complicated because they need to make digests with two
+            specific properties:
+            <ul>
+                <li>It must be impossible to use the digest to figure out the original message</li>
+                <li>It must be hard to make a different message that has the same digest</li>
+            </ul>
+            In order to get these properties, hash functions do things like always produce a hash
+            that is the same size, regardless of the size of the message, or making a completely
+            different digest for messages that are almost exactly the same.
```

Comment:
I'd actually bring up the diffusion/confusion idea into your bulleted list. I don't have the textbook in front of me, but this seems like one of the important properties of a hash function - that even small changes in the input will produce (large) changes in the output. This makes it possible to use hashes to verify the integrity of downloaded files.